Rubberduck.CodeAnalysis/QuickFixes/RemoveEmptyIfBlockQuickFix.cs

```diff
@@ -7,25 +7,33 @@
 using Rubberduck.Parsing.Grammar;
 using Rubberduck.Parsing.Inspections.Abstract;
 using Rubberduck.Parsing.Rewriter;
-using Rubberduck.Parsing.VBA;
 
 namespace Rubberduck.Inspections.QuickFixes
 {
-    internal sealed class RemoveEmptyIfBlockQuickFix : QuickFixBase
+    public sealed class RemoveEmptyIfBlockQuickFix : QuickFixBase
     {
-        private readonly RubberduckParserState _state;
-
-        public RemoveEmptyIfBlockQuickFix(RubberduckParserState state)
+        public RemoveEmptyIfBlockQuickFix()
             : base(typeof(EmptyIfBlockInspection))
-        {
-            _state = state;
-        }
+        {}
 
-        public override void Fix(IInspectionResult result)
+        public override void Fix(IInspectionResult result, IRewriteSession rewriteSession)
         {
-            var rewriter = _state.GetRewriter(result.QualifiedSelection.QualifiedName);
-
-            UpdateContext((dynamic)result.Context, rewriter);
+            var rewriter = rewriteSession.CheckOutModuleRewriter(result.QualifiedSelection.QualifiedName);
+
+            switch (result.Context)
+            {
+                case VBAParser.IfStmtContext ifContext:
+                    UpdateContext(ifContext, rewriter);
+                    break;
+                case VBAParser.IfWithEmptyThenContext ifWithEmtyThenContext:
+                    UpdateContext(ifWithEmtyThenContext, rewriter);
+                    break;
+                case VBAParser.ElseIfBlockContext elseIfBlockContext:
+                    UpdateContext(elseIfBlockContext, rewriter);
+                    break;
+                default:
+                    throw new NotSupportedException(nameof(result.Context));
```

Comment:
shouldn't we use the type instead of the name here?

Reply:
That would make sense. To be honest, I was rather unsure which approach is the correct/more sensible here.